extension/test/e2e/credit-card-amount-update.spec.js

```diff
@@ -0,0 +1,214 @@
+    const creditCardDate = '03/30'
+    const creditCardCvc = '737'
+
+    try {
+      const baseUrl = config.getModuleConfig().apiExtensionBaseUrl
+      const clientKey = config.getAdyenConfig(adyenMerchantAccount).clientKey
+      const browserTab = await browser.newPage()
+
+      // Step #1 - Create a payment session
+      // https://docs.adyen.com/online-payments/web-components#create-payment-session
+      paymentAfterCreateSession = await createSession(clientKey)
+      logger.debug(
+        'credit-card-amount-update::paymentAfterCreateSession:',
+        JSON.stringify(paymentAfterCreateSession)
+      )
+
+      // Step #2 - Setup Component
+      // https://docs.adyen.com/online-payments/web-components#set-up
+
+      initPaymentSessionResult = await initPaymentSession({
+        browserTab,
+        baseUrl,
+        clientKey,
+        paymentAfterCreateSession,
+        creditCardNumber,
+        creditCardDate,
+        creditCardCvc,
+      })
+      await waitUntil(
+        async () =>
+          await fetchNotificationInterfaceInteraction(
+            paymentAfterCreateSession.id
+          )
+      )
+
+      const notificationInteraction =
+        await fetchNotificationInterfaceInteraction(
+          paymentAfterCreateSession.id
+        )
+
+      // Step #3 - Update Amount
+      const { statusCode, updatedPayment } = await updateAmount(
+        notificationInteraction,
+        paymentAfterCreateSession
+      )
+      amountUpdatesResponse = JSON.parse(
+        updatedPayment.custom.fields.amountUpdatesResponse
+      )
+      amountUpdatesInterfaceInteractions =
+        updatedPayment.interfaceInteractions.filter(
+          (ii) => ii.fields.type === 'amountUpdates'
+        )
+      updatedAmountStatusCode = statusCode
+    } catch (err) {
+      logger.error('credit-card-amount-update::errors:', JSON.stringify(err))
+    }
+
+    assertCreatePaymentSession(
+      paymentAfterCreateSession,
+      initPaymentSessionResult
+    )
+    expect(updatedAmountStatusCode).to.equal(200)
+    expect(amountUpdatesResponse.status).to.equal('received')
+    expect(amountUpdatesInterfaceInteractions).to.have.lengthOf(1)
+  })
+
+  async function updateAmount(
+    notificationInteraction,
+    paymentAfterCreateSession
+  ) {
+    const notificationStr = notificationInteraction.fields.notification
+    const notificationJson = JSON.parse(notificationStr)
+    const paymentPspReference =
+      notificationJson.NotificationRequestItem.pspReference
+    const amountUpdatesRequestDraft = {
+      paymentPspReference,
+      amount: {
+        currency: 'EUR',
+        value: 1010,
+      },
+      reason: 'DelayedCharge',
+      reference: paymentAfterCreateSession.key,
+    }
+    const { body: paymentAfterReceivingNotification } =
+      await ctpClient.fetchById(
+        ctpClient.builder.payments,
+        paymentAfterCreateSession.id
+      )
+    const { statusCode, body: updatedPayment } = await ctpClient.update(
```

Comment:
How does this really work?
I don't see any waiting time here for Extension module to execute Amount Updates endpoint(Calling Adyen for payment).

Reply:
We didn't check the notification of amount updates all the time. In the past we did verify the response from API endpoint in integration test, which is now implemented in this e2e test.

I checked a bit the corresponding payment (paymennt ID : "f8802d2b-c234-43cd-8319-77afd0528ae4"), found that amount update notification was received in notification module. However it was not inserted into interactionInterface. Hence we have no way to obtain the notification in extension module even waiting function is implemented here, unless we check and fix it in the notification module webhook test.

Reply:
In the previous Integration tests, we were asserting the response from adyen without waiting for Notification.

Now, I see that you are waiting for Notification to come from Adyen and why don't we directly assert the response from Notification here?
Here, if we can assert Notification response then we don't need to explicitly run Webhook tests to assert the response.
WDYT? Do you see any problems?

Reply:
>but actually it doesn't work now due to issue when amount updates notification is received by notification module.

Do you know what is the issue now in the Notification module?

My point is, If Notification module processes the response from amount updates notification, Then we can assert it here only(In the e2e tests).

Also we don't need to write Webhook tests.

Reply:
The scope of this pull request is to migrate the existing integration test for amount update to an end-to-end test. Verifying notification of amount updates actually is additional part, which was not implemented in existing test, and is never the scope of this pull request.

We have already another ticket to ensure the notification of amount updates to be received and operate properly in notification module. Therefore, I would suggest to investigate and fix the mentioned issue in another featured branch, which belongs to that ticket. Also, In this next featured branch, it doesn't matter whether adding webhook test in notification module or adding assertion in this amount updates e2e test. It would be fine if you prefer to do assertion in e2e test after you fixed the issue.

Reply:
My question is more of, Can we do it or do you see any issues with e2e tests.
Because, I was thinking of eliminating Webhook tests itself and achieve these assertions in e2e tests.
Anyhow, I will take up from here. I will check the issues and try to fix the Notification module to get proper responses to assert.